src/hoc/withTheme.js

```diff
@@ -31,6 +35,22 @@ export default (Component: ReactClass<mixed>) => class extends React.Component {
   render() {
     const { theme } = this.state
 
-    return <Component theme={theme} {...this.props} />
+    // eslint-disable-next-line react/prop-types
+    let { innerRef } = this.props
+    let ref
+
+    if (!isStyledComponent(Component)) {
+      ref = innerRef
+      innerRef = undefined
+    }
+
+    return (
+      <Component
+        theme={theme}
+        {...this.props}
+        innerRef={innerRef}
+        ref={ref}
+      />
+    )
```

Comment:
I was trying to avoid the double-merge as a micro-optimisation, but it's definitely cleaner. Should I change it?